src/middlewared/middlewared/plugins/smb.py

```diff
@@ -809,17 +809,103 @@ class Config:
+        following ACE:
+
+        `owner@:full_set:fd:allow`
+
+        The actual ACL written on the path specified in the UI will vary for
+        homes shares depending on whether Active Directory is enabled. In all
+        cases, the ACL is written so that users have adequate permissions to
+        traverse to their home directory and the actual home directory is only
+        accessible by the user.
+
+        If an SMB admin group has been selected, then it will also be added to the
+        share's ACL.
+        """
+        if not default_perms:
+            return
+
+        acl = []
+        admin = None
+        smb = await self.middleware.call('smb.config')
+        if smb['admin_group']:
+            admin = await self.middleware.call('notifier.get_group_object', smb['admin_group'])
+            acl.append({
+                "tag": "GROUP",
+                "id": admin[2],
+                "type": "ALLOW",
+                "perms": {"BASIC": "FULL_CONTROL"},
+                "flags": {"BASIC": "INHERIT"}
+            })
+
+        acl.append({
+            "tag": "owner@",
+            "id": None,
+            "type": "ALLOW",
+            "perms": {"BASIC": "FULL_CONTROL"},
+            "flags": {"BASIC": "INHERIT"},
+        })
+
+        if not is_home:
+            acl.append({
+                "tag": "group@",
+                "id": None,
+                "type": "ALLOW",
+                "perms": {"BASIC": "FULL_CONTROL"},
+                "flags": {"BASIC": "INHERIT"},
+            })
+
+        elif await self.middleware.call('activedirectory.get_state') != 'DISABLED':
+            acl.extend([
+                {
+                    "tag": "group@",
+                    "id": None,
+                    "type": "ALLOW",
+                    "perms": {"BASIC": "MODIFY"},
+                    "flags": {'DIRECTORY_INHERIT': True, 'INHERIT_ONLY': True, 'NO_PROPAGATE_INHERIT': True}
+                },
+                {
+                    "tag": "everyone@",
+                    "id": None,
+                    "type": "ALLOW",
+                    "perms": {"BASIC": "TRAVERSE"},
+                    "flags": {"BASIC": "NOINHERIT"}
+                },
+            ])
+
+        else:
+            acl.extend([
+                {
+                    "tag": "group@",
+                    "id": None,
+                    "type": "ALLOW",
+                    "perms": {"BASIC": "MODIFY"},
+                    "flags": {"BASIC": "NOINHERIT"}
+                },
+                {
+                    "tag": "everyone@",
+                    "id": None,
+                    "type": "ALLOW",
+                    "perms": {"BASIC": "TRAVERSE"},
+                    "flags": {"BASIC": "NOINHERIT"}
+                },
+            ])
+
+        job = await self.middleware.call('filesystem.setacl', path, acl, {'recursive': True, 'traverse': True})
+        await job.wait()
```

Comment:
Not sure we want to wait here since apply_default_perms is not a job and it can timeout? Perhaps just let it do its thing in background?